tsconfig.json

```diff
@@ -13,6 +14,7 @@
 		"strictBindCallApply": true,
 		"strictFunctionTypes": true,
 		"sourceMap": true,
+		"esModuleInterop": true,
 		"jsx": "react",
```

Comment:
I was getting the following error in my LinkSelector.ts file when trying to import the 'open' module

`Module '"/Users/jkrl/Programs/joplin/CliClient/node_modules/open/index"' can only be default-imported using the 'esModuleInterop' flag`

Here's a SO link about esModuleInterop: https://stackoverflow.com/questions/56238356/understanding-esmoduleinterop-in-tsconfig-file. When typescript compiles it compiles to CommonJS imports so as far as I could tell I needed 'esModuleInterop' to import my 'open' module using the require syntax.

For the es2020.string addition I needed that to use the 'matchAll' function in LinkSelector.ts as this is a newer function that typescript doesn't recognize without importing the newer string library

Lastly for dom and dom.iterable I was getting weird errors in parts of the joplin directories during tsc that I didn't even edit like this:

`ElectronClient/gui/NoteEditor/NoteBody/AceEditor/AceEditor.tsx(446,3): error TS2584: Cannot find name 'document'. Do you need to change your target library? Try changing the `lib` compiler option to include 'dom'.`

Adding both dom and dom.iterable fixed the problems.

Reply:
Would it be possible to make it work without the changes to tsconfig? (as it's likely to affect other things) You can use `const bla = require('bla')` if that helps, or have a look at @CalebJohn's pull request for other ways to get TypeScript to work with external modules: https://github.com/laurent22/joplin/pull/3375

Reply:
Alright, so when I use CommonJS imports instead, esModulesInterop is no longer necessary so I have removed that. Not sure how I missed that before.


As far as the imported libraries...

I don't think that's going to effect the rest of the program as all they're are doing is adding typescript functionality and not taking anything away. 'es2020.string' I think is going to be necessary because 'matchAll' is a new function that typescript doesn't recognize unless I include that import, and that's just the best function to use inside my LinkSelector file.

As far as dom and dom.iterable, I'm still getting errors at build time inside of inside of files related to TinyMCE unless I include them. Maybe you can build without those imports yourself and you would have a better insight into what the problem is? Because I didn't edit the files that are being errored on and have no idea why that's happening.